components/x-interaction/readme.md

```diff
@@ -108,3 +108,114 @@ export const BaseGreeting = ({greeting, actions}) => <div>
+
+It returns a function that, when called, resets the serialisation state, and returns all serialisation data collected from any components that were rendered in the meantime, in the format of a `<script>` tag that should be included in the rendered page. This function *must* be called once at the end of every request when `getInteractionSerialiser`, _after_ all `x-interaction` components have been rendered. Any component rendered after this function is called won't have its data included in the output.
+
+If you call `getInteractionSerialiser` once for multiple requests, it will collect data for components that aren't rendered in the current request, causing inflated output size and losing that data for the pages that need it. Calling it multiple times per request is a no-op.
+
+If you don't call the function returned by `getInteractionSerialiser` at the end of the request, the data collected during this request will be lost, and possibly returned in a future request. When calling it multiple times, all calls after the first will return empty data, and including these in the response will potentially erase the global variable it outputs on the client side.
+
+Here's a full example of using `getInteractionSerialiser`:
+
+```js
+import express from 'express';
+import { getInteractionSerialiser } from '@financial-times/x-interaction';
+import { Increment } from '@financial-times/x-increment';
+
+const app = express();
+
+app.get('/', (req, res) => {
+	const getInteractionData = getInteractionSerialiser();
+
+	res.send(`
+		${Increment({count: 1})}
+		${getInteractionData()}
+	`);
+});
+```
+
+#### Hydrating
+
+When rendered on the server side, components output an extra wrapper element, with a `data-x-dash-id` attribute. This is used by the `x-interaction` runtime to identify the component in serialisation data, and attach the correct instance to it on the client side. You can pass this in as the `id` property when rendering the component; otherwise, an identifier will be randomly generated.
+
+`x-interaction` exports a function `hydrate`. This should be called on the client side. It inspects the global serialisation data on the page, uses the identifiers to find the wrapper elements, and calls `render` from your chosen `x-engine` client-side runtime to render component instances into the wrappers.
+
+Before calling `hydrate`, you must first `import` any `x-interaction` components that will be rendered on the page. The components register themselves with the `x-interaction` runtime when imported; you don't need to do anything with the imported component. This will also ensure the component is included in your client-side bundle.
+
+Because `hydrate` expects the wrappers to be present in the DOM when called, it should be called in an event such as `DOMContentLoaded`.
```

Comment:
Maybe just say "The `hydrate` function should be called after the `DOMContentLoaded` event" and link off to MDN for info about that?